tests/baselines/reference/privateNameDuplicateField.errors.txt

```diff
@@ -1,11 +1,657 @@
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(331,20): error TS2300: Duplicate identifier '#foo'.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(332,16): error TS2300: Duplicate identifier '#foo'.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(337,20): error TS2300: Duplicate identifier '#foo'.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(338,16): error TS2300: Duplicate identifier '#foo'.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(343,20): error TS2300: Duplicate identifier '#foo'.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(344,20): error TS2300: Duplicate identifier '#foo'.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(357,9): error TS2804: Duplicate identifier '#foo'. Static and instance elements cannot share the same private name.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(363,9): error TS2804: Duplicate identifier '#foo'. Static and instance elements cannot share the same private name.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(370,13): error TS2804: Duplicate identifier '#foo'. Static and instance elements cannot share the same private name.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(376,13): error TS2804: Duplicate identifier '#foo'. Static and instance elements cannot share the same private name.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(382,16): error TS2300: Duplicate identifier '#foo'.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(387,20): error TS2300: Duplicate identifier '#foo'.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(388,16): error TS2300: Duplicate identifier '#foo'.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(399,20): error TS2300: Duplicate identifier '#foo'.
+tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts(400,20): error TS2300: Duplicate identifier '#foo'.
 
 
-==== tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts (1 errors) ====
-    class A {
-        #foo = "foo";
-        #foo = "foo";
-        ~~~~
+==== tests/cases/conformance/classes/members/privateNames/privateNameDuplicateField.ts (83 errors) ====
+    function Field() {
+    
+        // Error
+        class A_Field_Field {
+            #foo = "foo";
+            #foo = "foo";
+            ~~~~
 !!! error TS2300: Duplicate identifier '#foo'.
+        }
+    
+        // Error
+        class A_Field_Method {
+            #foo = "foo";
+            ~~~~
+!!! error TS2300: Duplicate identifier '#foo'.
+            #foo() { }
+            ~~~~

```

Comment:
It's surprising that there's any inconsistency here. Any idea why that's happening?

Reply:
I am unsure of the exact reasons. Some duplication errors come from the binder some from the checker. I think this has something to do with overloads.  The effect is that if the duplication is  field-method the error comes from the binder (it also has info about the other class element). While duplication for method-field comes from the checker (and does not have related info about the other class element). 

This is the reason I massively expanded the duplication tests, as since the errors can come from two different places and order matters, I wanted to make sure I don't miss any corner cases.

I think the duplicate detection code might be due for a revamp. It was quite confusing to have the error to come from different places if the order differs. I tried to just make the changes needed for private static-instance duplication duplication,  as such a revamp was not really in scope for this issue.